docs/content/guides/developer/app-examples/e2e-counter.mdx

```diff
@@ -1,176 +1,363 @@
 ---
 title: Distributed Counter
+hide_table_of_contents: true
 ---
 
-This example walks you through building a basic distributed counter app, covering the full end-to-end flow connecting your Move code to your client app. The app allows you to create counters that anyone can increment, but only the owner can reset. This example assumes you already have a React App set with dApp Kit, and it's required `Providers` as described in [Client App with Sui TypeScript SDK](../first-app/client-tssdk.mdx).
+This example walks you through building a basic distributed counter app, covering the full end-to-end flow of building your Sui Move module and connecting it your React Sui Dapp. The app allows you to create counters that anyone can increment, but only the owner can reset. 
 
-{@include: ../../../snippets/info-pnpm-required.mdx}
+The guide is split into two parts:
 
-If haven't followed Client App with Sui TypeScript SDK, run the following command in a terminal or console to scaffold a new app:
+1. [Smart Contracts](#smart-contracts): The Move code set ups the `Counter` structure and logic.
+1. [Frontend](#frontend): A UI that enables users to create, increment, and reset `Counter` objects.
 
-```bash
-pnpm create @mysten/dapp --template react-client-dapp
+## What you need
+
+Before getting started, make sure you have:
+
+- [Installed the latest version of Sui](../getting-started/sui-install.mdx).
+
+## Smart contracts {#smart-contracts}
+
+In this part of the guide, you will write the Move contracts that create, increment, and reset counters. The first step is to [set up a Move package](../first-app/write-package.mdx) for storing your Move modules. In this guide, the Move package will be labeled: `counter`.
+
+### `Counter` struct
+
+```move
+public struct Counter has key {
+  id: UID,
+  owner: address,
+  value: u64
+}
 ```
+- The `Counter` type stores the address of its `owner`, its current `value`, and its own `id`.
 
-or
 
-```bash
-yarn create @mysten/dapp --template react-client-dapp
+### Creating `Counter`
+
+```move
```

Comment:
Use the code inject plugin for code. I can add this at the end if you prefer. This one would be {@inject: examples/<path-to-file>#fun=create noComments}